File: facedetection.cpp
#include <opencv2/opencv.hpp>
#include <opencv2/objdetect.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
    // Load the Haar Cascade for face detection
    CascadeClassifier faceCascade;
    if (!faceCascade.load("haarcascade_frontalface_default.xml")) {
        cout << "Error loading face cascade file. Make sure the path is correct." << endl;
        return -1;
    }

    // Open the default video camera
    VideoCapture cap(0);
    if (!cap.isOpened()) {
        cout << "Error opening video stream or file" << endl;
        return -1;
    }

    // Loop to capture frames from the camera
    while (true) {
        Mat frame;
        cap >> frame;

        // Check if frame is empty
        if (frame.empty()) {
            cout << "No frame captured from camera." << endl;
            break;
        }

        // Convert to grayscale for face detection
        Mat gray;
        cvtColor(frame, gray, COLOR_BGR2GRAY);
        equalizeHist(gray, gray);

        // Detect faces
        vector<Rect> faces;
        faceCascade.detectMultiScale(gray, faces);

        // Draw rectangles around faces
        for (size_t i = 0; i < faces.size(); i++) {
            rectangle(frame, faces[i], Scalar(255, 0, 0), 2);
        }

        // Show the output frame
        imshow("Face Detection", frame);

        // Exit if the user presses 'q'
        char c = (char)waitKey(10);
        if (c == 'q' || c == 27) {
            break;
        }
    }

    // Release the camera or video capture object
    cap.release();
    destroyAllWindows();
    return 0;
}

File: facedetection.py

#you need to have OpenCV installed. You can install it with the following command
# pip install opencv-python


#You need to download the Haar Cascade XML file for face detection. You can get it from OpenCV's GitHub repository.
#Download haarcascade_frontalface_default.xml and place it in the same directory as your script.

import cv2

# Load the pre-trained Haar Cascade classifier for face detection
face_cascade = cv2.CascadeClassifier('haarcascade_frontalface_default.xml')

# Start capturing video from the default webcam
cap = cv2.VideoCapture(0)

if not cap.isOpened():
    print("Error: Could not open video.")
    exit()

while True:
    # Read a frame from the webcam
    ret, frame = cap.read()
    
    if not ret:
        print("Error: Could not read frame.")
        break

    # Convert the frame to grayscale, which is needed for Haar Cascade
    gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)

    # Detect faces in the grayscale image
    faces = face_cascade.detectMultiScale(gray, scaleFactor=1.1, minNeighbors=5, minSize=(30, 30))

    # Draw rectangles around detected faces
    for (x, y, w, h) in faces:
        cv2.rectangle(frame, (x, y), (x+w, y+h), (255, 0, 0), 2)

    # Display the frame with detected faces
    cv2.imshow('Face Detection', frame)

    # Exit the loop if 'q' is pressed
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

# Release the video capture object and close all OpenCV windows
cap.release()
cv2.destroyAllWindows()
